misc/Prime.c

```diff
@@ -2,8 +2,17 @@
 #include <math.h>
 
 int isPrime(int x) {
-    for (int i = 2; i < sqrt(x); i++) {
-        if (x%i == 0)
+    if (x == 2) {
+        return 1;
+    }
+    if (x < 2 || x % 2 == 0) {
+        return 0;
+    }
+    
+    double squareRoot = sqrt(x);
+    
+    for (int i = 3; i < squareRoot; i += 2) {
```

Comment:
the condition should be ``i <= squareRoot``.

because in the existing code,  If ``x`` is **9**, then ``squareRoot`` becomes **3**, and the condition in the for loop fails & ``return 1;`` is executed implying that **9** is a prime number